### PR TITLE
Handling borrowingvault external full debt repay

### PR DESCRIPTION
**Pull request by DaigaroCota**
This pull request addresses Item #-43 of the Three Sigma security audit.

### CODE REVIEW

**Review by 0x73696d616f — APPROVED** [review]
Looks good. I added some comments below, please check. Additionally, maybe the `mulDiv` could be used in `lockedAssets`, `_computeFreeAssets`.

File: packages/protocol/src/vaults/borrowing/BorrowingVault.sol [hunk 1]
```diff
@@ -29,6 +29,8 @@ import {Math} from "openzeppelin-contracts/contracts/utils/math/Math.sol";
 import {BaseVault} from "../../abstracts/BaseVault.sol";
 import {VaultPermissions} from "../VaultPermissions.sol";
 
+import {console} from "forge-std/console.sol";
```
Comment (0x73696d616f) [inline packages/protocol/src/vaults/borrowing/BorrowingVault.sol:32]: console.log is here, probably better to remove before merging

File: packages/protocol/src/vaults/borrowing/BorrowingVault.sol [hunk 2]
```diff
@@ -544,7 +515,8 @@ contract BorrowingVault is BaseVault {
     uint256 debtShares = _debtShares[borrower];
     uint256 debt = convertToDebt(debtShares);
 
-    uint256 baseUserMaxBorrow = ((assets * maxLtv * price) / (1e18 * 10 ** decimals()));
+    // uint256 baseUserMaxBorrow = ((assets * maxLtv * price) / (1e18 * 10 ** decimals()));
+    uint256 baseUserMaxBorrow = assets.mulDiv(maxLtv.mulDiv(price, 1e18), 10 ** decimals());
```
Comment (0x73696d616f) [inline packages/protocol/src/vaults/borrowing/BorrowingVault.sol:519]: I think the math is a bit different now, there could be precision loss. This is the new formula, based on the new code
```solidity
assets*(maxLtv*price/1e18)/10**decimals()
```
doing instead
```solidity
assets.mulDiv(maxLtv*price, 10 ** decimals()*1e18);
```
would prevent precision loss
the risk would be `maxLtv*price` overflow

File: packages/protocol/src/vaults/borrowing/BorrowingVault2.sol [hunk 3]
```diff
@@ -423,7 +392,8 @@ contract BorrowingVault2 is BaseVault {
     uint256 debtShares = _debtShares[borrower];
     uint256 debt = convertToDebt(debtShares);
 
-    uint256 baseUserMaxBorrow = ((assets * maxLtv * price) / (1e18 * 10 ** decimals()));
+    // uint256 baseUserMaxBorrow = ((assets * maxLtv * price) / (1e18 * 10 ** decimals()));
+    uint256 baseUserMaxBorrow = assets.mulDiv(maxLtv.mulDiv(price, 1e18), 10 ** decimals());
```
Comment (0x73696d616f) [inline packages/protocol/src/vaults/borrowing/BorrowingVault2.sol:396]: same comment as above

**Review by pedrovalido — APPROVED** [review]

**Pull request merged**